Tools/LoadNMs.linq

```diff
@@ -0,0 +1,305 @@
+<Query Kind="Program">
+  <Reference>&lt;RuntimeDirectory&gt;\netstandard.dll</Reference>
+  <Reference>&lt;RuntimeDirectory&gt;\System.Net.Http.dll</Reference>
+  <NuGetReference>Microsoft.Identity.Client</NuGetReference>
+  <NuGetReference>Newtonsoft.Json</NuGetReference>
+  <NuGetReference>UKHO.FileShareAdminClient</NuGetReference>
+  <NuGetReference>UKHO.WeekNumberUtils</NuGetReference>
+  <Namespace>Microsoft.Identity.Client</Namespace>
+  <Namespace>System.Net.Http</Namespace>
+  <Namespace>System.Net.Http.Headers</Namespace>
+  <Namespace>System.Threading.Tasks</Namespace>
+  <Namespace>UKHO.FileShareAdminClient</Namespace>
+  <Namespace>UKHO.FileShareClient</Namespace>
+  <Namespace>System.Globalization</Namespace>
+  <Namespace>UKHO.WeekNumberUtils</Namespace>
+  <Namespace>UKHO.FileShareAdminClient.Models</Namespace>
+  <Namespace>UKHO.FileShareClient.Models</Namespace>
+</Query>
+
+const string BUSINESS_UNIT = "MaritimeSafetyInformation";
+const string PRODUCT_TYPE_ATTRIBUTE_VALUE = "Notices to Mariners";
+
+private readonly Func<DateTime, DateTime> WEEKLY_NMS_RETENTION_PERIOD = d => d.AddDays(800);
+private readonly Func<DateTime, DateTime> DAILY_NMS_RETENTION_PERIOD = d => d.AddDays(15);
+
+private readonly bool REPLACE_ALL_DATA_IN_BUSINESS_UNIT = false;
+
+private async Task Main()
+{
+	var configLive = new
+	{
+		BaseAddress = @"https://files.admiralty.co.uk/",
+		ClientId = "30cc156d-765f-43ea-8c84-a97d5b8ce852",
+		TenantId = "9134ca48-663d-4a05-968a-31a42f0aed3e",
+		AuthUrl = "https://login.microsoftonline.com/9134ca48-663d-4a05-968a-31a42f0aed3e/oauth2/v2.0/authorize"
+	};
+	var configQA = new
+	{
+		BaseAddress = @"https://fss-qa-webapp.azurewebsites.net",
+		ClientId = "805be024-a208-40fb-ab6f-399c2647d334",
+		TenantId = "9134ca48-663d-4a05-968a-31a42f0aed3e",
+		AuthUrl = "https://login.microsoftonline.com/9134ca48-663d-4a05-968a-31a42f0aed3e/oauth2/v2.0/authorize"
+	};
+	var configDev = new
+	{
+		BaseAddress = @"https://fss-dev-webapp.azurewebsites.net/",
+		ClientId = "3ec9403b-8299-4a90-8ecc-07ee6642109f",
+		TenantId = "9134ca48-663d-4a05-968a-31a42f0aed3e",
+		AuthUrl = "https://login.microsoftonline.com/9134ca48-663d-4a05-968a-31a42f0aed3e/oauth2/v2.0/authorize"
+	};
+
+	var config = configQA;
+
+	var fssClient = new FileShareApiAdminClient(new UserAgentClientFactory(), config.BaseAddress,
+		new AuthTokenProvider(config.ClientId, config.TenantId, config.AuthUrl));
+
+	if (REPLACE_ALL_DATA_IN_BUSINESS_UNIT)
+	{
+		await ExpireAllDataInBusinessUnit(fssClient, BUSINESS_UNIT);
+	}
+
+	await ProcessDailies(fssClient, @"\\hadsprddom04.business.ukho.gov.uk\DFS\Prod_Prod\nms\FILES\NMOutput\Dailies");
```

Comment:
I would prefer that these internal paths were not visible in a public repo

Reply:
I completely agree, so have deferred to storing this file in the "Support Engineer Toolbox".

Please see https://github.com/UKHO/supportengineer-toolbox/pull/62